### PR TITLE
OP-729 Update project to Java 8 and new time objects

### PR DESCRIPTION
**Pull request by dbmalkovsky**
See OP-729

Sorry but there are lots of changes; at least all the tests pass.

I also changed the jar version to 0.2 so as not to impact the UI development still using version 0.1

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -5,56 +5,47 @@
...
     <artifactId>openhospital-api</artifactId>
-    <version>0.0.1</version>
+    <version>0.0.2</version>
     <name>openhospital-api</name>
-    <description>Openhospital Spring Boot REST API </description>
+    <description>Open Hospital Spring Boot REST API</description>
 
     <properties>
-        <oh.version>1.11.0</oh.version>
+        <oh.version>1.11.1</oh.version>
```
Comment (mwithi) [inline pom.xml:18]:
```suggestion
        <oh.version>1.11.2</oh.version>
```

File: src/main/java/org/isf/disease/rest/DiseaseController.java [hunk 2]
```diff
@@ -243,15 +243,15 @@ public DiseaseController(DiseaseBrowserManager diseaseManager, DiseaseMapper dis
...
 			throw new OHAPIException(new OHExceptionMessage(null, "duplicated disease description for the same disease type", OHSeverityLevel.ERROR), HttpStatus.INTERNAL_SERVER_ERROR);
 		}
-		
+
 		if (diseaseManager.newDisease(disease) != null) {
         	return ResponseEntity.status(HttpStatus.CREATED).body(diseaseDTO);
-        } else {
-        	throw new OHAPIException(new OHExceptionMessage(null, "disease is not created!", OHSeverityLevel.ERROR), HttpStatus.INTERNAL_SERVER_ERROR);
         }
+        throw new OHAPIException(new OHExceptionMessage(null, "disease is not created!", OHSeverityLevel.ERROR), HttpStatus.INTERNAL_SERVER_ERROR);
```
Comment (mwithi) [inline src/main/java/org/isf/disease/rest/DiseaseController.java:254]: maybe indentation?

File: pom.xml (hunk 1 above)
Comment (dbmalkovsky) [reply pom.xml:18]: @mwithi I don't think I can do that as the main Core repo still has 1.11.1 in it.

File: src/main/java/org/isf/disease/rest/DiseaseController.java (hunk 2 above)
Comment (dbmalkovsky) [reply src/main/java/org/isf/disease/rest/DiseaseController.java]: I reformatted the entire method.

File: pom.xml (hunk 1 above)
Comment (mwithi) [reply pom.xml:18]: true... but I'm merging soon 1.11.2

**Review by mwithi — APPROVED** [review]
LGTM. Anyway I would ask a review to "ui" frontend to know if they can adopt this new J8 version

**Comment by dbmalkovsky** [comment]
Because there is a jar version change they can take it anytime they want and is convenient for them.

**Comment by mwithi** [comment]
> Because there is a jar version change they can take it anytime they want and is convenient for them.

yes, but if they can adopt it now with small changes on their side it is better to keep the full stack aligned IMHO and avoid the two components diverge too much

**Review by almeida1492 — APPROVED** [review]

**Comment by mwithi** [comment]
Thanks a lot guys!

**Pull request merged**